src/instanceMethods/close.js

```diff
@@ -117,12 +118,13 @@ const unsetWeakMaps = (obj) => {
   }
 }
 
-const triggerOnAfterClose = (onAfterClose) => {
-  if (onAfterClose !== null && typeof onAfterClose === 'function') {
-    setTimeout(() => {
+const triggerOnAfterCloseAndDispose = (instance, onAfterClose) => {
+  setTimeout(() => {
+    if (onAfterClose !== null && typeof onAfterClose === 'function') {
       onAfterClose()
-    })
-  }
+    }
+    disposeSwal(instance)
```

Comment:
The `disposeSwal()` executes after the `onAfterClose()`. If the `onAfterClose()` contains a new Swal, then the inner properties of the new Swal get disposed. I added a new test for that in https://github.com/sweetalert2/sweetalert2/pull/1655/commits/9bed120a28d80e346bda7c638f6f578bbec05dea

Reply:
Good catch and thanks a lot for providing a test case! A failing test case worth thousands of words in GitHub comments :)

The provided test case is not failing with the check I made in the latest commit: https://github.com/sweetalert2/sweetalert2/pull/1655/commits/4510d138ab31beca73292d405b8b5abade33d365